pkg/action/uninstall.go

```diff
@@ -61,12 +62,15 @@ func (u *Uninstall) Run(name string) (*release.UninstallReleaseResponse, error)
 	}
 
 	if u.DryRun {
-		// In the dry run case, just see if the release exists
 		r, err := u.cfg.releaseContent(name, 0)
-		if err != nil {
+		switch {
+		case err == nil:
+			fallthrough
+		case u.IgnoreNotFound && errors.As(err, &driver.ErrReleaseNotFound):
+			return &release.UninstallReleaseResponse{Release: r}, nil
+		default:
```

Comment:
IMHO, I think it would be better to not use the result of `r, err := u.cfg.releaseContent(name, 0)` iff `err != nil`. On an error, `r` is likely not well constructed. Furthermore, the [non-dry run logic](https://github.com/helm/helm/blob/ea9b823e674567b6e7046f0cc6a1bdc6a45f40a0/pkg/action/uninstall.go#L82-L84) returns `nil`, which I suspect should be replicated here.

ie. the logic should be:

```suggestion
		switch {
		case err == nil:
			return &release.UninstallReleaseResponse{Release: r}, nil
		case u.IgnoreNotFound && errors.As(err, &driver.ErrReleaseNotFound):
			return nil, nil
		default:
		        return &release.UninstallReleaseResponse{}, err
```

Reply:
That said, I perhaps wouldn't use a `switch` statement. I think subjectively an `if` might be easier to read:

```
if err != nil {
    if u.IgnoreNotFound && errors.As(err, &driver.ErrReleaseNotFound) {
        return nil, nil
    }
    return &release.UninstallReleaseResponse{}, err
}

return &release.UninstallReleaseResponse{Release: r}, nil
```

Reply:
(side comment) There seems to be an awkwardness in the function contract here. `Run()` is expected to return information about the uninstalled release (or error). But that can't happen when `u.IgnoreNotFound` is true.

It seems `u.IgnoreNotFound` logic should be outside of the uninstall action (but that is a refactor for another day). And today, if a client supplies `u.IgnoreNotFound = true`, the client should expect that a release may not be returned.

Reply:
I've changed the `switch` by a `if` and the return

Reply:
Thanks for the review @gjenkins8

Reply:
Any chance to get another review?